Create another offer if one gets taken

klochowicz opened the pull request:
See details in commit messages.

klochowicz committed:
Broadcast value of current_order instead of hard-coding to None

klochowicz committed:
Generate a new offer after previous is taken with the same parameters

There is no reason for waiting for interaction with the UI/automation - if the
parameters (price, funding rate) were OK for one taker, they should be OK
for another one.

klochowicz committed:
Be more explicit about Position parameter in Order constructor

klochowicz committed:
Simplify model::Order constructor

Don't return Result, as nothing can fail inside it.

thomaseizinger commented on daemon-tests/tests/happy_path.rs:
```diff
@@ -175,6 +175,37 @@ async fn taker_takes_order_and_maker_rejects() {
 }
 
+#[tokio::test]
+async fn another_offer_is_automatically_created_aftertaker_takes_order() {
```
```suggestion
async fn another_offer_is_automatically_created_after_taker_takes_order() {
```

thomaseizinger commented on daemon-tests/tests/happy_path.rs:
```diff
@@ -175,6 +175,37 @@ async fn taker_takes_order_and_maker_rejects() {
+    // TODO: Why is this needed? For the cfd stream it is not needed
+    is_next_none(taker.order_feed()).await.unwrap();
+
+    maker.publish_order(dummy_new_order(Position::Short)).await;
```
We might want to rename this to something like `set_offer_parameters`?

klochowicz replied:
sounds good, changed this and also changed `NewOrder` to `OfferParams`

klochowicz committed:
Rename "publish order" to "set offer params"

Disambiguate in regards to the nature of the action done on the maker.
Setting offer params keeps creating new offers with the same parameters until
the parameters are changed or maker goes offline.

klochowicz committed:
Set offer params in tests using a dedicated maker actor system method

Avoid bypassing dedicated method by sending a message directly to the actor.

thomaseizinger reviewed (approved):
I like this change but it might be controversial so please gather some more approvals before merging :)

bonomat reviewed (approved):
Let's give this a try +1
